### DISCUSSION TITLE
seek help! About the use of activationlock

### DISCUSSION THREAD

**Discussion by axing21cn** [micromdm/micromdm#878] [Q&A] [answered]
I am a newcomer to microcomdm. Starting microcomdm is easy, but there are some MDM commands that I really don't know how to use.

For example, in tools/commands, there is no example of settings, and I think activating ActivationLock has been unsuccessful.

I tried to submit a command like this, but it seems to be unsuccessful.

{

"request_type":"Settings",

"settings":[{

"item":"MDMOptions",

"mdm_options":{

"ActivationLockAllowedWhileSupervised":true

}

}],

"udid":"11111"

}

What should I do now? I just want to prevent my phone from getting out of supervision. Do I need to modify anything else in the profile file? I am currently using the default configuration of MicroMDM

**Comment by axing21cn** [comment]
Execute the above command, and the content returned by micromdm is as follows

{
	"payload":{
		"command_uuid":"152892f6-2849-48d3-88d0-666b845a6225",
		"command":{
			"request_type":"Settings",
			"settings":[{
					"item":"MDMOptions",
					"identifier":null,
					"mdm_options":{
						"ActivationLockAllowedWhileSupervised":true
					},
					"configuration":null
				}]
		}
	}
}

**Comment by xastherion** [comment]
Hi axing,
Is your device (iphone) enroll?
Your device muss shine if you run a "mdmctl get devices" command.
How send you this command? Normally, you need the command + UUID or Serial and it work if the device is enrolled.
I recommend you to try before another commands like "restart device" to see if micromdm and the device are running OK.
I hope it help you.
Xast

**Reply by axing21cn** [reply]
Thank you for Xast's response. I have confirmed that the device has been registered in Dep and MDM, and just confirmed the device settings.

<? xml version="1.0" encoding="UTF-8"?>

<! DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" " http://www.apple.com/DTDs/PropertyList-1.0.dtd ">

<plist version="1.0">

<dict>

<key>CommandUUID</key>

<string>4c9d26b7-bc58-4cd5-a117-189306de684b</string>

<key>QueryResponses</key>

<dict>

<key>IsActivationLockEnabled</key>

<false/>

<key>IsMDMLostModeEnabled</key>

<false/>

<key>IsSupervised</key>

<true/>

<key>MDMOptions</key>

<dict>

<key>ActivationLockAllowedWhileSupervised</key>

<true/>

</dict>

</dict>

<key>Status</key>

<string>Acknowledged</string>

<key>UDID</key>

<string>944ec82c02a5d355fe2133c15f4d3c4e1bba9f31</string>

</dict>

</plist>




ActivationLockAllowedWhileSupervised in MDMOptions has taken effect

But IsActivationLockEnabled is still false, how can I configure it?

According to the current situation, click on "Exit Remote Management" on your phone, follow the prompts to erase all data, and activate directly without any assistance. This is not what I want.

Looking at Apple's documentation, after the activation lock takes effect and the device is erased, MDM authorization or the original ABM registrant account is required to activate the device. I just want to achieve this.

**Comment by axing21cn** [comment]
![image](https://user-images.githubusercontent.com/23007117/233106185-08816803-7af5-4623-831a-d60cef00a468.png)
<key>IsActivationLockEnabled</key>
I don't know how to set this state to true

**Reply by jessepeterson** [reply, marked as answer]
I'd take a look at @mboylan's timely blog on this here: https://blog.kandji.io/how-to-manage-activation-lock :)

This involves MDM commands such as https://developer.apple.com/documentation/devicemanagement/get_the_bypass_code_for_activation_lock  and DEP endpoints such as https://developer.apple.com/documentation/devicemanagement/activation_lock_a_device

**Reply by jessepeterson** [reply]
Also, join us over in the [MacAdmins Slack](https://www.macadmins.org) in #micromdm channel. There's more of us there to discuss things like this — I suspect not as many folks see the Discussions here.

**Reply by axing21cn** [reply]
https://developer.apple.com/documentation/devicemanagement/activation_lock_a_device
I have also read this document. I just don't quite understand the authentication process of OAuth. This method directly interacts with Apple servers. What I want to know is whether Micromdm provides relevant interfaces for convenient calling?

**Reply by axing21cn** [reply]
Thank you. The problem has been resolved.

**Reply by 62218** [reply]
> https://developer.apple.com/documentation/devicemanagement/activation_lock_a_device I have also read this document. I just don't quite understand the authentication process of OAuth. This method directly interacts with Apple servers. What I want to know is whether Micromdm provides relevant interfaces for convenient calling?

@axing21cn  can you tell me how to do it

**Reply by mkdirnow** [reply]
Could you please share your experience? Thanks in advance.

**Reply by punon-tech** [reply]
> Thank you. The problem has been resolved.

@axing21cn can you please post here how you resolved the issue. We are facing similar issue

**Reply by jessepeterson** [reply]
OAuth for DEP endpoints is standard DEP authentication, just like other endpoints. Take a look at [NanoDEP web proxy](https://github.com/micromdm/nanodep/blob/main/docs/operations-guide.md#reverse-proxy) to be able  use those endpoints with your ABM or ASM accounts.
